internal/hcsoci/layers.go

```diff
@@ -90,21 +99,45 @@ func MountContainerLayers(layerFolders []string, guestRoot string, uvm *uvm.Util
 			}
 			err = uvm.AddVSMB(layerPath, "", options)
 			if err == nil {
-				vsmbAdded = append(vsmbAdded, layerPath)
+				wcowLayersAdded = append(wcowLayersAdded, layerPath)
 			}
 		} else {
 			uvmPath := ""
-			_, uvmPath, err = uvm.AddVPMEM(filepath.Join(layerPath, "layer.vhd"), true) // UVM path is calculated. Will be /tmp/vN/
-			if err == nil {
-				vpmemAdded = append(vpmemAdded,
-					vpMemEntry{
-						hostPath: filepath.Join(layerPath, "layer.vhd"),
-						uvmPath:  uvmPath,
-					})
+			hostPath := filepath.Join(layerPath, "layer.vhd")
+
+			var fi os.FileInfo
+			fi, err = os.Stat(hostPath)
+			if err == nil && fi.Size() > maxPMEMSize {
+				// Too big for PMEM. Add on SCSI instead (at /tmp/S<C>/<L>).
+				// As SCSI, we have to grant access first.
+				if err = wclayer.GrantVmAccess(uvm.ID(), hostPath); err == nil {
```

Comment:
Just noticed this. For a followup, but the GrantVmAccess calls can be moved to the add SCSI implementation rather than the callers doing it (here and for scratch)

Reply:
Addressed in https://github.com/Microsoft/hcsshim/pull/364